src/doc/src/reference/resolver.md

```diff
@@ -0,0 +1,415 @@
+`1.0.0-beta` is published, then a requirement `foo = "1.0.0-alpha"` will allow
+updating to the `beta` version. Beware that pre-release versions can be
+unstable, and as such care should be taken when using them. Some projects may
+choose to publish breaking changes between pre-release versions. It is
+recommended to not use pre-release dependencies in a library if your library
+is not also a pre-release. Care should also be taken when updating your
+`Cargo.lock`, and be prepared if a pre-release update causes issues.
+
+The pre-release tag may be separated with periods to distinguish separate
+components. Numeric components will use numeric comparison. For example,
+`1.0.0-alpha.4` will use numeric comparison for the `4` component. That means
+that if `1.0.0-alpha.11` is published, that will be chosen as the greatest
+release. Non-numeric components are compared lexicographically.
+
+[`cargo install`]: ../commands/cargo-install.md
+
+### Version metadata
+
+SemVer has the concept of "version metadata" with a plus in the version, such
+as `1.0.0+21AF26D3`. This metadata is usually ignored, and should not be used
+in a version requirement. You should never publish multiple versions that
+differ only in the metadata tag (note, this is a [known issue] with
+[crates.io] that currently permits this).
+
+[known issue]: https://github.com/rust-lang/crates.io/issues/1059
+[crates.io]: https://crates.io/
+
+## Other constraints
+
+Version requirements aren't the only constraint that the resolver considers
+when selecting and unifying dependencies. The following sections cover some of
+the other constraints that can affect resolution.
+
+### Features
+
+The resolver resolves the graph as-if the [features] of all [workspace]
+members are enabled. This ensures that any optional dependencies are available
+and properly resolved with the rest of the graph when features are added or
+removed with the `--features` command-line flag. The actual features used when
+*compiling* a crate will depend on the features enabled on the command-line.
+
+Dependencies are resolved with the union of all features enabled on them. For
+example, if one package depends on the [`im`] package with the [`serde`
+dependency] enabled and another package depends on it with the [`rayon`
+dependency] enabled, then `im` will be built with both features enabled, and
+the `serde` and `rayon` crates will be included in the resolve graph. If no
+packages depend on `im` with those features, then those optional dependencies
+will be ignored, and they will not affect resolution.
+
+The resolver will skip over versions of packages that are missing required
+features. For example, if a package depends on version `^1` of [`regex`] with
+the [`perf` feature], then the oldest version it can select is `1.3.0`,
+because versions prior to that did not contain the `perf` feature. Similarly,
+if a feature is removed from a new release, then packages that require that
```

Comment:
Is it worth calling out the case of making an optional dependency non-optional, which is an easy way to accidentally remove a feature?